src/components/ImportOnyxState/index.native.tsx

```diff
@@ -55,22 +33,25 @@ export default function ImportOnyxState({setIsLoading}: ImportOnyxStateProps) {
             return;
         }
 
+        rollbackOngoingRequest();
         setIsLoading(true);
         readOnyxFile(file.uri)
             .then((fileContent: string) => {
                 const transformedState = cleanAndTransformState<OnyxValues>(fileContent);
                 const currentUserSessionCopy = {...session};
                 setPreservedUserSession(currentUserSessionCopy);
                 setShouldForceOffline(true);
-                Onyx.clear(KEYS_TO_PRESERVE).then(() => {
-                    applyStateInChunks(transformedState).then(() => {
-                        setIsUsingImportedState(true);
-                        Navigation.navigate(ROUTES.HOME);
-                    });
-                });
+                return processStateImport(transformedState);
+            })
+            .then(() => {
+                setIsUsingImportedState(true);
+                Navigation.navigate(ROUTES.HOME);
             })
             .catch(() => {
```

Comment:
```suggestion
            .catch((error) => {
                console.error('Error importing state:', error);
```

to align between platform files